server/test/unit/server/agent/direct/test_agent.py

```diff
@@ -28,8 +28,44 @@ def test_capabilities(self):
         self.assertEqual(agent.profile, Profile)
 
 
+class NotFound(Exception):
+    pass
+
+
 class TestConsumerCapability(TestCase):
 
+    @patch('pulp.server.agent.direct.pulpagent.Queue')
+    def test_delete_queue(self, queue):
+        url = 'test-url'
+        name = 'test-queue'
+
+        # test
+        agent = PulpAgent()
+        agent.delete_queue(url, name)
+
+        # validation
+        queue.assert_called_once_with(name)
+        queue = queue.return_value
+        queue.purge.assert_called_once_with(url)
+        queue.delete.assert_called_once_with(url)
+
+    @patch('pulp.server.agent.direct.pulpagent.Queue')
+    @patch('pulp.server.agent.direct.pulpagent.NotFound', NotFound)
+    def test_delete_queue_not_found(self, queue):
+        url = 'test-url'
+        name = 'test-queue'
+        queue.return_value.purge.side_effect = NotFound
+
+        # test
+        agent = PulpAgent()
```

Comment:
If the code is broken then an exception will be raised here. consider putting it in a try/except block and in the except block putting a self.fail(). That will cause it to be marked as a failure which is more accurate than an Error.